diff --git a/cmake/onnxruntime_config.h.in b/cmake/onnxruntime_config.h.in
index 2aef9dcf20..e3ea767401 100644
--- a/cmake/onnxruntime_config.h.in
+++ b/cmake/onnxruntime_config.h.in
@@ -22,5 +22,5 @@
 #cmakedefine HAS_UNUSED_BUT_SET_VARIABLE
 #cmakedefine HAS_UNUSED_VARIABLE
 #cmakedefine HAS_USELESS_CAST
-#cmakedefine ORT_BUILD_INFO u8"@ORT_BUILD_INFO@"
-#cmakedefine ORT_VERSION u8"@ORT_VERSION@"
+#cmakedefine ORT_BUILD_INFO "NO_BUILD_INFO"
+#cmakedefine ORT_VERSION "@ORT_VERSION@"
