diff --git a/folly/Exception.h b/folly/Exception.h
index 3b3118c61..db51d0fa3 100644
--- a/folly/Exception.h
+++ b/folly/Exception.h
@@ -53,11 +53,11 @@ inline std::system_error makeSystemErrorExplicit(int err, const char* msg) {
   return std::system_error(err, errorCategoryForErrnoDomain(), msg);
 }
 
-template <class... Args>
-std::system_error makeSystemErrorExplicit(int err, Args&&... args) {
-  return makeSystemErrorExplicit(
-      err, to<fbstring>(std::forward<Args>(args)...).c_str());
-}
+// template <class... Args>
+// std::system_error makeSystemErrorExplicit(int err, Args&&... args) {
+//   return makeSystemErrorExplicit(
+//       err, to<fbstring>(std::forward<Args>(args)...).c_str());
+// }
 
 inline std::system_error makeSystemError(const char* msg) {
   return makeSystemErrorExplicit(errno, msg);

